File: llm_generated_code/Gemini/Python/Level1/Q27.py
def is_equilibrium(forces):
    x_sum, y_sum, z_sum = 0, 0, 0
    for force in forces:
        x_sum += force[0]
        y_sum += force[1]
        z_sum += force[2]
    return x_sum == 0 and y_sum == 0 and z_sum == 0

def main():
    n = int(input())
    forces = []
    for _ in range(n):
        x, y, z = map(int, input().split())
        forces.append((x, y, z))
    print("YES" if is_equilibrium(forces) else "NO")

if __name__ == "__main__":
    main()

File: llm_generated_code/Gemini/CPP/Level1/Q27.cpp
#include <iostream>
#include <vector>

using namespace std;

struct Force {
    double x, y, z;
};

bool isEquilibrium(vector<Force> &forces) {
    double sumX = 0, sumY = 0, sumZ = 0;
    for (const Force &force : forces) {
        sumX += force.x;
        sumY += force.y;
        sumZ += force.z;
    }

    return abs(sumX) < 1e-9 && abs(sumY) < 1e-9 && abs(sumZ) < 1e-9;
}

int main() {
    int n;
    cin >> n;

    vector<Force> forces(n);
    for (int i = 0; i < n; i++) {
        cin >> forces[i].x >> forces[i].y >> forces[i].z;
    }

    if (isEquilibrium(forces)) {
        cout << "YES" << endl;
    } else {
        cout << "NO" << endl;
    }

    return 0;
}
